File: Fib1000_euler25.java
import java.math.*;

public class Fib1000_euler25{
	public static void main(String[] args) {
		
		long startTime = System.currentTimeMillis();

		for(int i = 100; i < 20000; i++){
			String s = String.valueOf(fib(i));
			//System.out.println(s);
			if(s.length() >= 1000){
				System.out.println(i);
				break;
			}
		}

		long stopTime = System.currentTimeMillis();

		System.out.println((stopTime - startTime) + " ms");

	}
	private static BigInteger fib(int n){
		BigInteger a = BigInteger.ONE;
		BigInteger b = BigInteger.ONE;

		int stevec = 2;
		BigInteger tp = BigInteger.ZERO;

		while(stevec < n){
			tp = b;
			b = a.add(b);
			a = tp;
			stevec++;
		}

		return b;
	}
}

File: Fib1000_euler25.py
import time



def fib(n):
	a = 1
	b = 1
	
	index = 2
	tp = 0

	while(index < n):
		tp = b
		b = a + b
		a = tp
		index+=1

	return b

start_time = time.time()
for i in range(100, 20000):
	c = fib(i)
	if(len(list(str(c))) >= 1000):
		print(i)
		break

print(time.time() - start_time)
